beacon-chain/state/state_trie_test.go

```diff
@@ -83,3 +103,138 @@ func TestInitializeFromProtoUnsafe(t *testing.T) {
 		})
 	}
 }
+
+func TestBeaconState_HashTreeRoot(t *testing.T) {
+	testState, _ := testutil.DeterministicGenesisState(t, 64)
+
+	type test struct {
+		name        string
+		stateModify func(*state.BeaconState) (*state.BeaconState, error)
+		error       string
+	}
+	initTests := []test{
+		{
+			name: "unchanged state",
+			stateModify: func(beaconState *state.BeaconState) (*state.BeaconState, error) {
+				return beaconState, nil
+			},
+			error: "",
+		},
+		{
+			name: "different slot",
+			stateModify: func(beaconState *state.BeaconState) (*state.BeaconState, error) {
+				if err := beaconState.SetSlot(5); err != nil {
+					return nil, err
+				}
+				return beaconState, nil
+			},
+			error: "",
+		},
+		{
+			name: "different validator balance",
+			stateModify: func(beaconState *state.BeaconState) (*state.BeaconState, error) {
+				val, err := beaconState.ValidatorAtIndex(5)
+				if err != nil {
+					return nil, err
+				}
+				val.EffectiveBalance = params.BeaconConfig().MaxEffectiveBalance - params.BeaconConfig().EffectiveBalanceIncrement
+				if err := beaconState.UpdateValidatorAtIndex(5, val); err != nil {
+					return nil, err
+				}
+				return beaconState, nil
+			},
+			error: "",
+		},
+	}
+
+	var err error
+	var oldHTR []byte
+	for _, tt := range initTests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
One more good thing would be to add would be that the generic ssz hash tree root method also gives the same result.

Reply:
Good idea, added!